src/catalog/table_catalog.cpp

```diff
@@ -341,20 +344,30 @@ std::unique_ptr<catalog::Schema> TableCatalog::InitializeSchema() {
   database_id_column.AddConstraint(
       catalog::Constraint(ConstraintType::NOTNULL, not_null_constraint_name));
 
-  std::unique_ptr<catalog::Schema> table_catalog_schema(new catalog::Schema(
-      {table_id_column, table_name_column, database_id_column}));
+  // namespace column
+  auto table_namespace_column = catalog::Column(
+      type::TypeId::VARCHAR, max_name_size, "table_namespace", false);
+  table_namespace_column.AddConstraint(
+      catalog::Constraint(ConstraintType::NOTNULL, not_null_constraint_name));
+
+  std::unique_ptr<catalog::Schema> table_catalog_schema(
+      new catalog::Schema({table_id_column, table_name_column,
+                           database_id_column, table_namespace_column}));
 
   return table_catalog_schema;
 }
 
 /*@brief   insert a tuple about table info into pg_table
  * @param   table_oid
+ * @param   table_namespace
  * @param   table_name
  * @param   database_oid
  * @param   txn     TransactionContext
  * @return  Whether insertion is Successful
  */
-bool TableCatalog::InsertTable(oid_t table_oid, const std::string &table_name,
+bool TableCatalog::InsertTable(oid_t table_oid,
+                               const std::string &table_namespace,
```

Comment:
New arguments should be added last and assigned default values in order to avoid misbehaviors in previous tests.

Reply:
I disagree. `namespace` needs to come before `table_name`. It also should **not** have a default value.

Reply:
The reason why I put namespace (and session name for some other commits) after all the arguments and with a default value is because that some tests will call the function directly without going through connection. In that case, namespace should be default and session name should be empty. Is that not appropriate? Should I make those two values without default values and explicitly set them in the tests?